adapters/DuAd/DuAdAdapter/GADDuAdNativeAd.m

```diff
@@ -0,0 +1,354 @@
+    _nativeAd.delegate = nil;
+}
+
+- (void)loadNativeAdImages {
+    [_loadedImages removeAllObjects];
+    if (_nativeAdImageAdLoaderOptions.disableImageLoading) {
+        // Set scale as 1 since DuAd does not provide image scale.
+        GADNativeAdImage *nativeAdImage =
+        [[GADNativeAdImage alloc] initWithURL:[NSURL URLWithString:_nativeAd.imgeUrl] scale:1];
+        if (nativeAdImage) {
+            _images = @[ nativeAdImage ];
+        }
+        _icon = [[GADNativeAdImage alloc] initWithURL:[NSURL URLWithString:self->_nativeAd.iconUrl] scale:1];
+        [self nativeAdImagesReady];
+    } else {
+        // Load icon and cover image, and notify the connector when completed.
+        [self loadCoverImage];
+        [self loadIcon];
+    }
+}
+
+- (void)loadCoverImage {
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
+        NSData *imageData = [NSData dataWithContentsOfURL:[NSURL URLWithString:self->_nativeAd.imgeUrl]];
+        UIImage *image = [UIImage imageWithData:imageData];
+        if (image) {
+            GADNativeAdImage *nativeAdImage = [[GADNativeAdImage alloc] initWithImage:image];
+            if (nativeAdImage) {
+                self->_images = @[ nativeAdImage ];
+            }
+        }
+        [self completedLoadingNativeAdImage:GADNativeAdCoverImage];
+    });
+}
+
+- (void)loadIcon {
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
+        NSData *iconData = [NSData dataWithContentsOfURL:[NSURL URLWithString:self->_nativeAd.iconUrl]];
+        UIImage *icon = [UIImage imageWithData:iconData];
+        if (icon) {
+            self->_icon = [[GADNativeAdImage alloc] initWithImage:icon];
+        }
+        [self completedLoadingNativeAdImage:GADNativeAdIcon];
+    });
+}
+
+- (void)completedLoadingNativeAdImage:(NSString *)image {
+    __block BOOL loadedAllImages = NO;
+    dispatch_sync(_lockQueue, ^{
+        [self->_loadedImages addObject:image];
+        loadedAllImages = [self->_loadedImages isEqual:self->_nativeAdImages];
+    });
+    
+    if (!loadedAllImages) {
+        return;
+    }
+    
+    if (_images && _icon) {
+        [self nativeAdImagesReady];
+        return;
+    }
+    
+    id<GADMAdNetworkAdapter> strongAdapter = self->_adapter;
+    id<GADMAdNetworkConnector> strongConnector = self->_connector;
+    NSString *errorMessage = [[NSString alloc] initWithFormat:@"Unable to load native ad image."];
+    [strongConnector adapter:strongAdapter didFailAd:GADDUErrorWithDescription(errorMessage)];
+}
+
+- (void)nativeAdImagesReady {
+    id<GADMAdNetworkAdapter> strongAdapter = self->_adapter;
+    id<GADMAdNetworkConnector> strongConnector = self->_connector;
+    [strongConnector adapter:strongAdapter didReceiveMediatedNativeAd:self];
+}
+
+#pragma mark - GADMediatedNativeAd
+
+- (id<GADMediatedNativeAdDelegate>)mediatedNativeAdDelegate {
+    return self;
+}
+
+#pragma mark - GADMediatedNativeAppInstallAd
```

Comment:
Media view is now a required asset for native ads. If media view is not supported, you can return an `UIImageView` that shows the main image in it.

Please see this MoPub adapter implementation for more detail: https://github.com/googleads/googleads-mobile-ios-mediation/blob/6ed94c724a6e8ebf42219cbeb578292844669101/adapters/MoPub/MoPubAdapter/MoPubAdapterMediatedNativeAd.m#L197

Also, please make sure to always download the main image even if the `disableImageLoading` option is set to `YES` so that it can be shown in the media view.